File: packages/java/parser-jvm-plugin-nonnull/src/test/java/com/vaadin/hilla/parser/plugins/nonnull/nullable/NullableEndpoint.java
package com.vaadin.hilla.parser.plugins.nonnull.nullable;


import jakarta.persistence.Id;
import jakarta.persistence.Version;

@Endpoint
public class NullableEndpoint {

    public NullableFieldModel nullableFieldModel(NullableFieldModel nullableFieldModel) {
        return nullableFieldModel;
    }

    public static class NullableFieldModel {
        @Id
        public String id;
        @Version
        public Long version;
    }
}

File: packages/java/typescript-generator/src/test/resources/com/vaadin/hilla/parser/plugins/nonnull/nullable/snapshots/NullableEndpoint.ts
import { EndpointRequestInit } from "@vaadin/hilla-frontend";
import type NullableFieldModel from "./com/vaadin/hilla/parser/plugins/nonnull/nullable/NullableEndpoint/NullableFieldModel.js";
import client from "./connect-client.default.js";
async function nullableFieldModel(nullableFieldModel: NullableFieldModel | undefined, init?: EndpointRequestInit): Promise<NullableFieldModel | undefined> { return client.call("NullableEndpoint", "nullableFieldModel", { nullableFieldModel }, init); }
export { nullableFieldModel };
